カテゴリー管理画面一覧表示実装

kensansansansan opened the pull request:
## チケットのリンク
https://gizumo.backlog.com/view/GIZFE-1129

## やったこと
storeファイルのmodulesファイルのcategory.jsファイルを追加
pagesファイルのCategoriesファイルのList.vueをファイル追加

テストケースでの、

カテゴリーを押すとカテゴリーの一覧に飛び表示される
最新のものが一番上に来るように順番を変更する（降順で表示）
カテゴリー一覧画面とテキスト入力欄が表示されている
問題が起きた場合APIから返されたエラーメッセージを表示

を追加しました！

## やらないこと
なし

## テスト
https://gizumo.backlog.com/view/GIZFE-1131

## 特にレビューをお願いしたい箇所
なしと記載したいのですが、category.jsファイルが、いらない記述が多いい気がするので、そこを見ていただけるとありがたいです。

kensansansansan committed:
add : カテゴリー管理画面表示追加テストケース

add : _store → modules → 「category.js」ファイル追加  pages → Categories → 「List.vue」ファイル追加

テストケースでの、カテゴリーを押すとカテゴリーの一覧に飛び表示される仕様と、最新のものが一番上に来るように順番を変更する（降順で表示）と、カテゴリー一覧画面とテキスト入力欄が表示されていることと、問題が起きた場合APIから返されたエラーメッセージを表示を追加しました。

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+<template>
+  <div class="categories-list-mask">
+    <div class="categories-list-mask-common">
```
[imo]
classの指定をCategoryPostコンポーネントに移動すればdivタグを削除できそうです！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/09a2235c0e7d02f6fa995d605318dc4f7c5f2ee0

お疲れ様です！
コメントありがとうございます！

divタグ消去し、classの指定をCategoryPostコンポーネントに移動しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+        @handle-submit="handleSubmit"
+      />
+    </div>
+    <div>
```
[nits]
こちらのdivタグ不要であれば削除をお願いします！

kensansansansan replied:
上記に記入しているのですが、List.vueのdivタグを消去しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+      theads: ['カテゴリー名'],
+      category: '',
+      errorMessage: '',
+      doneMessage: '',
```
[must]
`errorMessage`はstateでも管理されていると思うので、コンポーネントからは削除し、stateで管理するように修正をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/e7fdd489221809660b2fced7aaa0e2a87e30e194

List.vueのerrorMessageを消し修正しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+      category: '',
+      errorMessage: '',
+      doneMessage: '',
+      disabled: false,
```
[must]
`disabled`はAPI通信の状態を管理するプロパティだと思うので、こちらもstoreで管理しましょう！
（API通信はstoreで行うので、API通信の状態もstoreで管理した方が管理しやすいため）

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/43ab0fd383a8f09840fe2a38f946307019e266ed#diff-0f11587495060fd3ac11ac18788660b32ed2b9ca0dd445cbfa771710c3eedf13


disabled: falseをstoreで管理できるよう記述し、修正しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+      this.doneMessage = '';
+    },
+    handleSubmit() {
+    },
```
[must]
`handleSubmit`は関数の中身が空っぽなので、削除をお願いします！
将来的に使う想定かもしれないですが、基本的には今回の実装に必要なもの以外は差分に含めるべきではありません。

理由としてはレビュワーのノイズにもなりますし、もしかしたら使用されない可能性もあるためです。

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/a06074d43c5018823ec21975b6a73688184b8a0d

List.vueのhandleSubmit関数を消し修正しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+          this.categoryList = this.$store.state.categoryList;
+        }).catch(() => {
+          // console.log(err);
+        });
```
[ask]
こちらのthenメソッドとcatchメソッドを追加した意図を教えてください！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/cc8f9801c373f88257d3538fed76e43634dcd8ad

あんまり深く考えず、henメソッドはAPIを取得した後画面に表示させようと思って記述していました、catchメソッドは
エラーが出た時にしっかりと、出るのか、と言うのを、書いていたんですが、修正して行くうちに、いらない記述だったので、修正しました。

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,87 @@
+      theads: ['カテゴリー名'],
+      category: '',
+      errorMessage: '',
+      doneMessage: '',
```
[must]
今回の実装で`doneMessage`は不要だと思うので、削除をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/4791819ed2052249dca375a0a7610e5b49505a83

List.vueファイルのdoneMessageを消し修正しました！

keisuke3 commented on src/js/_router/index.js:
```diff
@@ -52,6 +56,17 @@ const router = new VueRouter({
       component: Home,
     },
+    {
+      path: '/category',
```
[must]
pathの指定はBacklogにある仕様書通りにお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/2b875063f6d6313ce6192f9887cca6cf741f9718

仕様書通りに書き換えたのですが下記のnamespaced: trueと同じコミットにしてしまいました。

変更内容は、js/_roter/index.jsファイルのpath: ''の記述を、path: '/categories',に変更し、
js/_helpers/routeLinksArray.jsファイルのpath: '/categoye'の記述をpath: '/categories'に変更し修正しました！

keisuke3 commented on src/js/_store/modules/category.js:
```diff
@@ -0,0 +1,31 @@
+import axios from '@Helpers/axiosDefault';
+
+export default {
```
[must]
`namespaced: true`を指定するようにお願いします！
`namespaced: true`の役割も調べて記載してください！

kensansansansan replied:
modules/category.jsファイルにnamespaced: trueを追加し修正しました！

役割としては、Vuex 2.1以降で使え、namespaced: trueを追加することで、自動的に自分が属している名前をつけてくれるようにする役割を持っています！

kensansansansan replied:
すみません、https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/2b875063f6d6313ce6192f9887cca6cf741f9718と同じコミットしてしましました。

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/commit/2b875063f6d6313ce6192f9887cca6cf741f9718

URLが上記と一緒になってしまいました、すみません。

keisuke3 reviewed (commented):
コメントしました！

kensansansansan committed:
src/components/pages/Categories/List.vueファイルのclassの指定をCategoryPostコンポーネントに移動した、src/components/pages/Categories/List.vueのdivタグ不要だったので削除しました。

kensansansansan committed:
src/components/pages/Categories/List.vueファイルのコンポーネントから削除

kensansansansan committed:
src/components/pages/Categories/List.vueファイルのdisabled: falseをstoreで管理できるよ移動させ修正

kensansansansan committed:
pages/Categories/List.vueファイルのhandleSubmitを消し修正

kensansansansan committed:
pages/Categories/List.vueファイルのdoneMessageを消し修正

kensansansansan committed:
modules/category.jsファイルにnamespaced: trueを追加し修正

kensansansansan committed:
src/js/_store/modules/category.jsファイルのnamespaced: trueを新たに記述したことと
src/js/_router/index.jsファイルのpath: '/categories'に変え、js/_helpers/routeLinKsファイルのパスをpath: '/categories'に変え修正

kensansansansan committed:
pages/Categories/List.vueファイルgetAllCategories部分を修正

keisuke3 commented on src/components/molecules/CategoryPost/index.vue:
```diff
@@ -81,7 +81,13 @@ export default {
 };
 </script>
 <style lang="scss" scoped>
+.categories-list-mask-common{
+  width: 40%;
+  padding-right: 8px;
+  border-right: 1px solid #eaeaea;
+}
 .category-management-post {
+  width: 39%;
```
[must]
こちらのスタイルに関してはmoleculesではなく、Pages/List.vueに書くようにお願いします！
moleculesは使い回すコンポーネントなので、コンポーネントの外に影響を与えるスタイルはなるべく適用したくないためです。（コンポーネントの外に影響を与えると、使用できる箇所が限定的になり汎用性が落ちてしまいます。 ）

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/f84804d06b7ef522ec14c5af8e08d5b0ab5c610c

moleculesに記述していたスタイルをpages/Categories/List.vuuファイルに記述し修正しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,68 @@
+    <app-category-list
+      :theads="theads"
+      :categories="categoriesList"
+      :delete-category-name="name"
```
[must]
`name`というプロパティはコンポーネント内に存在しないため、propsが渡せてなさそうです！
正しいと思う形に修正をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/e84a08085443e1181d8069da506299286d36774c

delete-category-name="name"消去の時に使い、まだ使わないと判断したので、消しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,68 @@
+    updateCategory(value) {
+      this.category = value;
+    },
+    clearMessage() {
+      this.errorMessage = '';
+      this.doneMessage = '';
+    },
```
[must]
data内に`errorMessage`と`doneMessage`がないため、こちらも不要だと思うので削除をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/ffa9682679c99ebc3b207124843f013ce7fe751d

pages/Categories/List.vuuファイルのerrorMessage、doneMessageを消去しました！

keisuke3 commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,68 @@
+  <div class="categories-list-mask">
+    <app-category-post
+      :category="category"
+      :error-message="errorMessage"
+      :done-message="doneMessage"
```
[must]
`errorMessage`と`doneMessage`は現在コンポーネント内に存在しないため、不要であれば削除をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/f54fa935c6b6a8a43322f093d61bf3691f8cf16d

pages/Categories/List.vuuファイルのerrorMessageとdoneMessage消去し、修正しました！

keisuke3 reviewed (commented):
コメントしました！

kensansansansan committed:
moleculesに記述していたスタイルをpages/Categories/List.vuuファイルに記述修正

kensansansansan committed:
pages/Categories/List.vuuファイルの:delete-category-name=nameを消す修正

kensansansansan committed:
pages/Categories/List.vuuファイルのclearMessageを消去しました

kensansansansan committed:
pages/Categories/List.vuuファイルのerrorMessageとdoneMessage消去修正

keisuke3 commented on src/components/molecules/CategoryPost/index.vue:
```diff
@@ -82,6 +82,7 @@ export default {
 </script>
 <style lang="scss" scoped>
 .category-management-post {
+  width: 39%;
```
[nits]
こちら使用されてなさそうなので削除をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/0de950900a04baa6b1b5adc1d1ae2ff38b8cc799

CategoryPost/index.vueファイルのwidth消去修正しました！

keisuke3 commented on src/components/molecules/CategoryPost/index.vue:
```diff
@@ -1,5 +1,5 @@
 <template>
-  <form @submit.prevent="addCategory">
+  <form class="categories-list-mask-common" @submit.prevent="addCategory">
```
[nits]
こちらも使用されてなさそうなので削除をお願いします！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/f92b0b094c94752ec29b96bd314c8ca9b8733d3d

form class="categories-list-mask-common"は使っていたので、CategoryPost/index.vueのclass="categories-list-mask-common"は消して、Categories/List.vueのapp-category-postに:class="{ 'categories-list-mask-common': true }"と付与し修正しました！

keisuke3 reviewed (commented):
コメントしました！

kensansansansan committed:
CategoryPost/index.vueファイルのwidth消去修正

kensansansansan committed:
CategoryPost/index.vueのclassを消去、Categories/List.vueのapp-category-postに:classを付与修正

ryo-kihara commented:
[must]

カテゴリ一覧画面を開いた時にいくつかコンソールエラーが出てそうなので、対応をお願いします！

常にコンソールにエラーが出てないかはチェックするようにしましょう！
画面で操作している時に問題がなくても、潜在的なエラーがある場合もあるので

ryo-kihara commented on src/js/_store/modules/category.js:
```diff
@@ -0,0 +1,33 @@
+  state: {
+    categoryList: [],
+    errorMessage: '',
+    disabled: false,
```
[nits]

こちらの state は使用していなさそうなので削除でお願いします

```suggestion
```

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/pull/252/commits/715943c07782e7673ccc64fdfacdd02a8e38377a

src/js/_store/modules/category.jsファイルのstate使用していないので消去修正しました！

ryo-kihara replied:
指摘箇所がわかりづらかったかもしれませんが、disabled state だけを赤く表現して削除してくださいという意味でした！
errorMessage state は消しても問題なさそうか改めて確認お願いします！

kensansansansan replied:
errorMessageを消しても特に問題は見つからなかったので、消去しました！

kensansansansan replied:
56f579c6df70b05b4a416de020fae233ce16eef6

src/js/_store/modules/category.jsファイルのserrorMessageを消しても特に問題は見つからなかったので、消去しました！

ryo-kihara commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,67 @@
+      :error-message="errorMessage"
+      :disabled="disabled"
+      :access="access"
+      :class="{ 'categories-list-mask-common': true }"
```
[imo]

常に true なのであれば、バインドさせずに `class` に直接文字列でクラス名を指定するで良いかなと思います！

kensansansansan replied:
3ccc7c88c7352c8c96797a9d936feecd9bba70dc

pages/Categories/List.vueの記述に新たにクラスを付与しました！

ryo-kihara commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,67 @@
+<style lang="scss" scoped>
+.categories-list-mask {
+  display: flex;
+  justify-content: space-between;
```
[must]

画面幅を広げると、間に余白ができてしまっているので、右側いっぱいにリストが表示されるように修正をお願いします！
（1600px程で確認しました）

<img width="1195" alt="スクリーンショット 2023-07-21 16 58 34" src="https://github.com/gizumo-education/gizumo_wiki_lesson/assets/48352529/d972fe50-80b8-41a8-9e9e-50b680f15b71">

kensansansansan replied:
3ccc7c88c7352c8c96797a9d936feecd9bba70dc

画面を広げて右側いっぱいにリストが表示されるように修正しました！

kensansansansan committed:
src/js/_store/modules/category.jsファイルのstate使用していないので消去修正

kensansansansan committed:
pages/Categories/List.vueの記述に新たにクラスを付与修正

kensansansansan committed:
pages/Categories/List.vueの記述に新たにクラスを付与修正

ryo-kihara commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,74 @@
+    return {
+      theads: ['カテゴリー名'],
+      category: '',
+      disabled: false,
```
[ask]

この `disabled` に関して、画面上でどのような動作確認を行いましたか？
もし動作確認ができていないのであれば、現時点では不要かなと思います！

補足ですが、disabled が true に変わるときがぱっと見なさそうです。
リアクティブプロパティとして定義するのであれば、値を変化させないと意味がないかなと思います！

kensansansansan replied:
disabledですが、仕様書通りの見た目にする為と（作成ボタンが青色になり文言が作成だけになる）、コンソールに出るエラーを無くす為に使用していました！

kensansansansan replied:
3f15d80b201035e41b6b61399ba80ec263729ebe

app-category-post６行目に記述してある:disabled="disabled"をfalseにする為に使用していました！

画面上でどのような動作確認は、仕様書通りの見た目にする為と（作成ボタンが青色になり文言が作成だけになる）、コンソールに出るエラーを無くす為に使用していました！

app-category-post６行目の:disabled="disabled"とdisabled: falseをエラーも起こらないので消しました！

ryo-kihara commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,74 @@
+      theads: ['カテゴリー名'],
+      category: '',
+      disabled: false,
+      categories: [],
+      list: [],
```
[ask]

`categories` と `list` を定義した理由は何かありますか？
使用されていなさそうだったので！

kensansansansan replied:
65f1f55281d349d1ea75ec1cef6df2e78abd9bff

categories と listを最初記述していた時の物で消していなかったです特に理由はありません！
また消しても問題が発生しなかったので、消去しました！

ryo-kihara commented on src/components/pages/Categories/List.vue:
```diff
@@ -0,0 +1,74 @@
+        :category="category"
+        :disabled="disabled"
+        :access="access"
+        @update-value="updateCategory"
```
[ask]

`@update-value="updateCategory"` の記述に対して行った動作確認は何かありますか？

kensansansansan replied:
552979d9f287c40094c6478886b10c4d39681d6a

特に動作確認は行っていませんでした、すみません！

kensansansansan replied:
https://github.com/gizumo-education/gizumo_wiki_lesson/commit/552979d9f287c40094c6478886b10c4d39681d6a

特に動作確認は行っていませんでした、すみません！
@update-value="updateCategory" の記述は消しました！

kensansansansan committed:
src/js/_store/modules/category.jsファイルのserrorMessageを消去修正

kensansansansan committed:
src/js/_store/modules/category.jsファイルのserrorMessageを消去修正

kensansansansan committed:
src/js/_store/modules/category.jsファイルのserrorMessageを消去修正

kensansansansan committed:
src/components/pages/Categories/List.vuecategories と listを消去修正

kensansansansan committed:
src/components/pages/Categories/List.vueの@update-value=updateCategory消去修正
”

kensansansansan committed:
src/components/pages/Categories/List.vueの:disabled=disabledとdisabled: false消去修正

ryo-kihara commented on src/js/_store/modules/category.js:
```diff
@@ -0,0 +1,31 @@
+      state.categoryList = [...payload.categories].reverse();
+    },
+    failRequest(state, { message }) {
+      state.errorMessage = message;
```
[must]

`state.errorMessage` の state が定義されていなさそうなので定義しましょう！

kensansansansan replied:
105bedd9b4c0cfd854bd44a53de610bae5839f39

src/js/_store/modules/category.jsファイルのstateにerrorMessage: '',を記述しました！

kensansansansan committed:
src/js/_store/modules/category.jsファイルerrorMessageを記述修正

kensansansansan committed:
src/js/_store/modules/category.jsファイルerrorMessageを記述修正

kensansansansan committed:
src/js/_store/modules/category.jsファイルerrorMessageを記述修正
